meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionClient.java

```diff
@@ -136,27 +139,63 @@ public void joinLeaderElectionParticipantPool(String leaderPath) {
   public void joinLeaderElectionParticipantPool(String leaderPath, LeaderInfo userInfo) {
     // TODO: create participant entry with info
     subscribeAndTryCreateLeaderEntry(leaderPath);
+
+    LeaderInfo participantInfo = new LeaderInfo(userInfo);
+    createParticipantInfo(leaderPath, participantInfo);
+  }
+
+  private void createParticipantInfo(String leaderPath, LeaderInfo participantInfo) {
+
+    try {
+      // try to create participant info entry, assuming leader election group node is already there
+      _metaClient.create(leaderPath + PARTICIPANTS_ENTRY_PARENT + _participant, participantInfo,
+          MetaClientInterface.EntryMode.EPHEMERAL);
+    } catch (MetaClientNodeExistsException ex) {
+      LOG.info("Already joined pool {}.", leaderPath);
+      return;
+    } catch (MetaClientNoNodeException ex) {
```

Comment:
This nested structure is very unclear. Would suggest to use if check whether exist and create the root dir instead of let exception handle it.

Reply:
If we do check and only create if no root node, and then create actual leader node, we can still get MetaClientNoNodeException when creating leader node.